### PR TITLE
refactor `InferenceSlicer` overlap logic

### PR DESCRIPTION
**Pull request by SkalskiP**
# Description

- Deprecate the `overlap_ratio_wh` argument
- Introduce the `overlap_wh` argument
- Rename the `overlap_filter_strategy` argument to `overlap_filter`

## Type of change

Please delete options that are not relevant.

-   [ ] Bug fix (non-breaking change which fixes an issue)
-   [x] New feature (non-breaking change which adds functionality)
-   [x] This change requires a documentation update

## Docs

-   [x] Docs updated? What were the changes:

### CODE REVIEW

File: supervision/detection/tools/inference_slicer.py [hunk 1]
```diff
@@ -193,10 +218,13 @@ def _generate_offset(
...
-            overlap_ratio_wh (Tuple[float, float]): A tuple representing the desired
-                overlap ratio for width and height between consecutive slices. Each
-                value should be in the range [0, 1), where 0 means no overlap and a
-                value close to 1 means high overlap.
+            overlap_ratio_wh (Optional[Tuple[float, float]]): A tuple representing the
+                desired overlap ratio for width and height between consecutive slices.
+                Each value should be in the range [0, 1), where 0 means no overlap and
+                a value close to 1 means high overlap.
+            overlap_wh (Optional[Tuple[int, int]]): A tuple representing the desired
+                overlap for width and height between consecutive slices. Each value
```
Comment (LinasKo) [inline supervision/detection/tools/inference_slicer.py:226]: Can we say it's in pixels? I'd be slightly confused about measurement units.

**Review by LinasKo — CHANGES_REQUESTED** [review]

File: supervision/detection/tools/inference_slicer.py [hunk 2]
```diff
@@ -226,3 +263,25 @@ def _generate_offset(
...
+                "Please provide only one of them."
+            )
+        if overlap_ratio_wh is not None:
+            if not (0 <= overlap_ratio_wh[0] < 1 and 0 <= overlap_ratio_wh[1] < 1):
+                raise ValueError(
+                    "Overlap ratios must be in the range [0, 1). "
+                    f"Received: {overlap_ratio_wh}"
+                )
+        if overlap_wh is not None:
+            if not (overlap_wh[0] > 0 and overlap_wh[1] > 0):
```
[the remark this answers is not in the document]
Comment (SkalskiP) [reply supervision/detection/tools/inference_slicer.py:284]: 💯

File: supervision/detection/tools/inference_slicer.py (hunk 1 above)
Comment (SkalskiP) [reply supervision/detection/tools/inference_slicer.py:226]: 💯

File: supervision/detection/tools/inference_slicer.py [hunk 3]
```diff
@@ -211,10 +239,19 @@ def _generate_offset(
         """
         slice_width, slice_height = slice_wh
         image_width, image_height = resolution_wh
-        overlap_ratio_width, overlap_ratio_height = overlap_ratio_wh
+        overlap_width = (
+            overlap_wh[0]
+            if overlap_wh is not None
+            else int(overlap_ratio_wh[0] * slice_width)
```
[the remark this answers is not in the document]
Comment (SkalskiP) [reply supervision/detection/tools/inference_slicer.py:245]: It is validated in `_validate_overlap` that is called in the `InferenceSlicer` constructor.

[the remark this answers is not in the document]
Comment (LinasKo) [reply supervision/detection/tools/inference_slicer.py:245]:
```
from typing import Optional, Tuple



def _validate_overlap(
    overlap_ratio_wh: Optional[Tuple[float, float]],
    overlap_wh: Optional[Tuple[int, int]],
) -> None:
    if overlap_ratio_wh is not None and overlap_wh is not None:
        raise ValueError(
            "Both `overlap_ratio_wh` and `overlap_wh` cannot be provided. "
            "Please provide only one of them."
        )
    if overlap_ratio_wh is not None:
        if not (0 <= overlap_ratio_wh[0] < 1 and 0 <= overlap_ratio_wh[1] < 1):
            raise ValueError(
                "Overlap ratios must be in the range [0, 1). "
                f"Received: {overlap_ratio_wh}"
            )
    if overlap_wh is not None:
        if not (overlap_wh[0] >= 0 and overlap_wh[1] >= 0):
            raise ValueError(
                "Overlap values must be greater than or equal to 0. "
                f"Received: {overlap_wh}"
            )
        
    print("nuh-uh! :D")
        
a = None
b = None

_validate_overlap(a, b)
```

[the remark this answers is not in the document]
Comment (SkalskiP) [reply supervision/detection/tools/inference_slicer.py:245]: eeeh true :)

[the remark this answers is not in the document]
Comment (SkalskiP) [reply supervision/detection/tools/inference_slicer.py:245]: Just added the logic to handle that case :)

**Comment by LinasKo** [comment]
Looks good based on the code alone. Need some help with tests tomorrow?

**Comment by SkalskiP** [comment]
@LinasKo here is my testing colab: https://colab.research.google.com/drive/1NC3MelyQVtiv7wt9E6U-hlqkqEzg-x5G?usp=sharing

File: supervision/detection/tools/inference_slicer.py [hunk 4]
```diff
@@ -73,23 +82,37 @@ class InferenceSlicer:
         not a multiple of the slice's width or height minus the overlap.
     """
 
+    @deprecated_parameter(
+        old_parameter="overlap_filter_strategy",
+        new_parameter="overlap_filter",
+        map_function=lambda x: x,
+        warning_message="`{old_parameter}` in `{function_name}` is deprecated and will "
+        "be remove in `supervision-0.27.0`. Use '{new_parameter}' "
```
Comment (LinasKo) [inline supervision/detection/tools/inference_slicer.py:90]: typo: removed

File: supervision/detection/tools/inference_slicer.py [hunk 5]
```diff
@@ -73,23 +82,37 @@ class InferenceSlicer:
...
+        overlap_wh: Optional[Tuple[int, int]] = None,
+        overlap_filter: Union[OverlapFilter, str] = OverlapFilter.NON_MAX_SUPPRESSION,
         iou_threshold: float = 0.5,
         thread_workers: int = 1,
     ):
-        overlap_filter_strategy = validate_overlap_filter(overlap_filter_strategy)
+        if overlap_ratio_wh is not None:
+            warn_deprecated(
+                "`overlap_ratio_wh` in `InferenceSlicer.__init__` is deprecated and "
+                "will be remove in `supervision-0.27.0`. Use `overlap_wh` instead."
```
Comment (LinasKo) [inline supervision/detection/tools/inference_slicer.py:106]: typo: removed

**Review by LinasKo — APPROVED** [review]
Very tiny typos, but otherwise I see no issues. Approving preemptively.

**Pull request merged**